File: SuperiorPackGroup/Code Classes/WeeklyProduction.vb
Public Class WeeklyProduction

    Public Sub New()

    End Sub

    Public Sub New(ByVal customerName As String, ByVal itemID As Integer, ByVal itemCode As String, ByVal itemDescription As String, ByVal requiredCases As Integer, ByVal requiredBags As Integer, ByVal producedMonday As Integer,
                   ByVal producedTuesday As Integer, ByVal producedWednesday As Integer, ByVal producedThursday As Integer, ByVal producedFriday As Integer, ByVal producedSaturday As Integer, ByVal producedSunday As Integer,
                   ByVal scheduledMonday As Integer, ByVal scheduledTuesday As Integer, ByVal scheduledWednesday As Integer, ByVal scheduledThursday As Integer, ByVal scheduledFriday As Integer, ByVal scheduledSaturday As Integer,
                   ByVal scheduledSunday As Integer, ByVal productionWeek As String)

        Me.CustomerName = customerName
        Me.ItemID = itemID
        Me.ItemCode = itemCode
        Me.ItemDescription = itemDescription
        Me.RequiredCases = requiredCases
        Me.RequiredBags = requiredBags
        Me.ProducedMonday = producedMonday
        Me.ProducedTuesday = producedTuesday
        Me.ProducedWednesday = producedWednesday
        Me.ProducedThursday = producedThursday
        Me.ProducedFriday = producedFriday
        Me.ProducedSaturday = producedSaturday
        Me.ProducedSunday = producedSunday
        Me.ScheduledMonday = scheduledMonday
        Me.ScheduledTuesday = scheduledTuesday
        Me.ScheduledWednesday = scheduledWednesday
        Me.ScheduledThursday = scheduledThursday
        Me.ScheduledFriday = scheduledFriday
        Me.ScheduledSaturday = scheduledSaturday
        Me.ScheduledSunday = scheduledSunday
        Me.ProductionWeek = productionWeek

    End Sub

    Public Sub New(ByVal customerName As String, ByVal itemID As Integer, ByVal itemCode As String, ByVal itemDescription As String, ByVal requiredCases As Integer, ByVal requiredBags As Integer, ByVal producedMonday As Integer,
                   ByVal producedTuesday As Integer, ByVal producedWednesday As Integer, ByVal producedThursday As Integer, ByVal producedFriday As Integer, ByVal producedSaturday As Integer,
                   ByVal producedSunday As Integer, ByVal productionWeek As String)

        Me.CustomerName = customerName
        Me.ItemID = itemID
        Me.ItemCode = itemCode
        Me.ItemDescription = itemDescription
        Me.RequiredCases = requiredCases
        Me.RequiredBags = requiredBags
        Me.ProducedMonday = producedMonday
        Me.ProducedTuesday = producedTuesday
        Me.ProducedWednesday = producedWednesday
        Me.ProducedThursday = producedThursday
        Me.ProducedFriday = producedFriday
        Me.ProducedSaturday = producedSaturday
        Me.ProducedSunday = producedSunday
        Me.ProductionWeek = productionWeek

    End Sub

    Public Overrides Function Equals(obj As Object) As Boolean

        If obj Is Nothing OrElse Me.GetType().Equals(obj.GetType()) = False Then
            Return False
        Else
            Dim WP As WeeklyProduction = CType(obj, WeeklyProduction)
            Return ItemID = WP.ItemID AndAlso ProductionWeek = WP.ProductionWeek
        End If

    End Function

    Public Property RequiredCases As Integer
    Public Property CustomerName As String
    Public Property ItemID As Integer
    Public Property ItemCode As String
        Public Property ItemDescription As String
        Public Property RequiredBags As Integer
        Public Property ProducedMonday As Integer
        Public Property ProducedTuesday As Integer
        Public Property ProducedWednesday As Integer
        Public Property ProducedThursday As Integer
        Public Property ProducedFriday As Integer
        Public Property ProducedSaturday As Integer
        Public Property ProducedSunday As Integer
        Public Property ScheduledMonday As Integer
        Public Property ScheduledTuesday As Integer
        Public Property ScheduledWednesday As Integer
        Public Property ScheduledThursday As Integer
        Public Property ScheduledFriday As Integer
        Public Property ScheduledSaturday As Integer
    Public Property ScheduledSunday As Integer
    Public Property ProductionWeek As String
    Public Property TotalProduced As Integer

    Public ReadOnly Property PercentageComplete As Double
        Get
            Return TotalProduced / RequiredCases
        End Get
    End Property

    Public ReadOnly Property TotalProducedWeek As Integer
        Get
            Return ProducedMonday + ProducedTuesday + ProducedWednesday + ProducedThursday + ProducedFriday + ProducedSaturday + ProducedSunday
        End Get
    End Property

End Class

File: SuperiorPackGroup/Code Classes/WeeklyProduction.cs
//INSTANT C# NOTE: Formerly VB project-level imports:
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;

namespace SuperiorPackGroup
{
	public class WeeklyProduction
	{

		public WeeklyProduction()
		{

		}

		public WeeklyProduction(string customerName, int itemID, string itemCode, string itemDescription, int requiredCases, int requiredBags, int producedMonday, int producedTuesday, int producedWednesday, int producedThursday, int producedFriday, int producedSaturday, int producedSunday, int scheduledMonday, int scheduledTuesday, int scheduledWednesday, int scheduledThursday, int scheduledFriday, int scheduledSaturday, int scheduledSunday, string productionWeek)
		{

			CustomerName = customerName;
			ItemID = itemID;
			ItemCode = itemCode;
			ItemDescription = itemDescription;
			RequiredCases = requiredCases;
			RequiredBags = requiredBags;
			ProducedMonday = producedMonday;
			ProducedTuesday = producedTuesday;
			ProducedWednesday = producedWednesday;
			ProducedThursday = producedThursday;
			ProducedFriday = producedFriday;
			ProducedSaturday = producedSaturday;
			ProducedSunday = producedSunday;
			ScheduledMonday = scheduledMonday;
			ScheduledTuesday = scheduledTuesday;
			ScheduledWednesday = scheduledWednesday;
			ScheduledThursday = scheduledThursday;
			ScheduledFriday = scheduledFriday;
			ScheduledSaturday = scheduledSaturday;
			ScheduledSunday = scheduledSunday;
			ProductionWeek = productionWeek;

		}

		public WeeklyProduction(string customerName, int itemID, string itemCode, string itemDescription, int requiredCases, int requiredBags, int producedMonday, int producedTuesday, int producedWednesday, int producedThursday, int producedFriday, int producedSaturday, int producedSunday, string productionWeek)
		{

			CustomerName = customerName;
			ItemID = itemID;
			ItemCode = itemCode;
			ItemDescription = itemDescription;
			RequiredCases = requiredCases;
			RequiredBags = requiredBags;
			ProducedMonday = producedMonday;
			ProducedTuesday = producedTuesday;
			ProducedWednesday = producedWednesday;
			ProducedThursday = producedThursday;
			ProducedFriday = producedFriday;
			ProducedSaturday = producedSaturday;
			ProducedSunday = producedSunday;
			ProductionWeek = productionWeek;

		}

		public override bool Equals(object obj)
		{

			if (obj == null || GetType().Equals(obj.GetType()) == false)
			{
				return false;
			}
			else
			{
				WeeklyProduction WP = (WeeklyProduction)obj;
				return ItemID == WP.ItemID && ProductionWeek == WP.ProductionWeek;
			}

		}

		public int RequiredCases {get; set;}
		public string CustomerName {get; set;}
		public int ItemID {get; set;}
		public string ItemCode {get; set;}
			public string ItemDescription {get; set;}
			public int RequiredBags {get; set;}
			public int ProducedMonday {get; set;}
			public int ProducedTuesday {get; set;}
			public int ProducedWednesday {get; set;}
			public int ProducedThursday {get; set;}
			public int ProducedFriday {get; set;}
			public int ProducedSaturday {get; set;}
			public int ProducedSunday {get; set;}
			public int ScheduledMonday {get; set;}
			public int ScheduledTuesday {get; set;}
			public int ScheduledWednesday {get; set;}
			public int ScheduledThursday {get; set;}
			public int ScheduledFriday {get; set;}
			public int ScheduledSaturday {get; set;}
		public int ScheduledSunday {get; set;}
		public string ProductionWeek {get; set;}
		public int TotalProduced {get; set;}

        public double PercentageComplete => TotalProduced / (double)RequiredCases;

        public int TotalProducedWeek => ProducedMonday + ProducedTuesday + ProducedWednesday + ProducedThursday + ProducedFriday + ProducedSaturday + ProducedSunday;

    }

}